### PR TITLE
add method for Updating kind and content type

### PR DESCRIPTION
**Pull request by ShatalovYaroslav**

### CODE REVIEW

File: src/main/java/org/ow2/proactive/catalog/rest/controller/CatalogObjectController.java [hunk 1]
```diff
@@ -132,6 +133,25 @@ public CatalogObjectMetadataList create(
...
+                            @ApiResponse(code = 403, message = "Permission denied"),
+                            @ApiResponse(code = 400, message = "Wrong specified parameters: at least one should be present") })
+    @RequestMapping(value = "/{name}", method = PUT)
+    @ResponseStatus(HttpStatus.OK)
+    public CatalogObjectMetadata updateObjectMetadata(
+            @ApiParam(value = "sessionID", required = false) @RequestHeader(value = "sessionID", required = false) String sessionId,
+            @PathVariable String bucketName, @PathVariable String name,
+            @ApiParam(value = "The new kind of an object", required = false) @RequestParam(value = "kind", required = false) Optional<String> kind,
+            @ApiParam(value = "The new content type of an object - MIME type", required = false) @RequestParam(value = "contentType", required = false) Optional<String> contentType)
+            throws UnsupportedEncodingException, NotAuthenticatedException, AccessDeniedException {
```
Comment (sonarqube-activeeon) [inline src/main/java/org/ow2/proactive/catalog/rest/controller/CatalogObjectController.java]: MINOR Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 2 issues
* ![MINOR][MINOR] 2 minor

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [CatalogObjectController.java#L108](https://github.com/ow2-proactive/catalog/blob/0228f9dbf98219065453a7196a14504e52798f84/src/main/java/org/ow2/proactive/catalog/rest/controller/CatalogObjectController.java#L108): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/integration-test/java/org/ow2/proactive/catalog/service/CatalogObjectServiceIntegrationTest.java [hunk 2]
```diff
@@ -128,6 +129,18 @@ public void testListCatalogObjectsInBucket() {
         assertThat(catalogObjects).hasSize(3);
     }
 
+    @Test
+    public void testUpdateObjectMetadata() {
+        CatalogObjectMetadata catalogObjectMetadata = catalogObjectService.updateObjectMetadata(bucket.getName(),
+                                                                                                "catalog1",
+                                                                                                Optional.of("updated-kind"),
```
Comment (sophiesongge) [inline src/integration-test/java/org/ow2/proactive/catalog/service/CatalogObjectServiceIntegrationTest.java]: what is this catalog1?

File: src/main/java/org/ow2/proactive/catalog/service/CatalogObjectService.java [hunk 3]
```diff
@@ -154,6 +156,25 @@ private String getFileMimeType(FileNameAndContent file) {
...
+        if (!kind.isPresent() && !contentType.isPresent()) {
+            throw new WrongParametersException("at least one parameter should be present");
+        }
+        CatalogObjectEntity catalogObjectEntity = catalogObjectRevisionEntity.getCatalogObject();
+        if (kind.isPresent()) {
+            catalogObjectEntity.setKind(kind.get());
+        }
+        if (contentType.isPresent()) {
+            catalogObjectEntity.setContentType(contentType.get());
+        }
```
Comment (MaelAudren) [inline src/main/java/org/ow2/proactive/catalog/service/CatalogObjectService.java]: replace conditions with optional by:
- kind.ifPresent( k -> catalogObjectEntity.setKind(k));
- contentType.ifPresent( ct -> catalogObjectEntity.setContentType(ct));

File: src/test/java/org/ow2/proactive/catalog/service/CatalogObjectServiceTest.java [hunk 4]
```diff
@@ -124,12 +126,110 @@ public void testCreateCatalogObject() {
...
+                                                                                        Optional.of("updated-kind"),
+                                                                                        Optional.of("updated-contentType"));
+
+        assertThat(catalogObject).isNotNull();
+        assertThat(catalogObject.getCommitMessage()).isEqualTo(COMMIT_MESSAGE);
+        assertThat(catalogObject.getContentType()).isEqualTo("updated-contentType");
+        assertThat(catalogObject.getKind()).isEqualTo("updated-kind");
+        assertThat(catalogObject.getName()).isEqualTo(NAME);
+        assertThat(catalogObject.getMetadataList()).isNotEmpty();
+        assertThat(catalogObject.getMetadataList()).hasSize(1);
```
Comment (MaelAudren) [inline src/test/java/org/ow2/proactive/catalog/service/CatalogObjectServiceTest.java]: You could add a test with the commit time, to ensure that the revision id is good.

**Pull request merged**